mobile-app/src/firebase/api.ts

```diff
@@ -96,73 +110,102 @@ export async function fetchHabitCompletionsForParticipant({
 }
 
 // FRIENDS
-export async function fetchFriends(): Promise<allUsersInfoT> {
-  const userId = atomStore.get(currentUserIdAtom);
-
+export async function fetchFriendIds({
+  userId,
+}: {
+  userId: string;
+}): Promise<string[]> {
+  const friendshipsCollection = collection(firestore, "friendships");
+  const q = query(
+    friendshipsCollection,
+    or(where("user1Id", "==", userId), where("user2Id", "==", userId)),
+  );
+  const myFriendshipSnap = await getDocs(q);
   const myFriendIds: string[] = [];
-  for (const friendshipId in mockFriendships) {
-    const friendship = mockFriendships[friendshipId];
+  myFriendshipSnap.forEach((doc) => {
+    const friendship = doc.data() as friendshipT;
     if (friendship.user1Id === userId) {
       myFriendIds.push(friendship.user2Id);
     } else if (friendship.user2Id === userId) {
       myFriendIds.push(friendship.user1Id);
     }
-  }
+  });
+  return myFriendIds;
+}
 
-  return Object.fromEntries(
-    Object.entries(mockUsers).filter(([userId]) =>
-      myFriendIds.includes(userId),
-    ),
-  );
+export async function fetchFriendData({
+  userId,
+}: {
+  userId: string;
+}): Promise<allUsersInfoT> {
+  const allFreindData: allUsersInfoT = {};
+  const myFriendIds = await fetchFriendIds({ userId });
+  for (const friendId of myFriendIds) {
+    const userDocRef = doc(firestore, "users", friendId);
+    const userDocSnap = await getDoc(userDocRef);
+    if (userDocSnap.exists()) {
+      const userData = userDocSnap.data() as userT;
+      allFreindData[friendId] = {
+        createdAt: userData.createdAt,
+        displayName: userData.displayName,
+        username: userData.username,
+        picture: userData.picture,
+      };
+    } else {
+      console.error(`No user found with ID: ${friendId}`);
+    }
+  }
```

Comment:
We could use an `in` query instead of for loop: https://firebase.google.com/docs/firestore/query-data/queries#in_not-in_and_array-contains-any

But we can only check a max of 30 things at once. So it would cause issues if the user has more than 30 friends.

So we could chunk it doing this type of thing: https://medium.com/dooboolab/solving-firestore-wherein-query-limitation-cb445f689da9

Or limit people to 30 friends max (at least for launch). Or just fetch first 30 and then load in more once they scroll to the bottom.

If you don't want to switch to use `in`, at least use `Promise.all` to fetch all of them in parallel: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
using the loop for now, make a linear task for the chunking and fetching like 10 at once then fetching more on scroll